### PR TITLE
Call on_error when non-partition related error happens

### PR DESCRIPTION
**Pull request by YijunXieMS**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py [hunk 1]
```diff
@@ -164,27 +164,33 @@ def _load_balancing(self):
...
+                self._handle_callback(self._error_handler, None, err)
                 # ownership_manager.get_checkpoints() and ownership_manager.claim_ownership() may raise exceptions
                 # when there are load balancing and/or checkpointing (checkpoint_store isn't None).
                 # They're swallowed here to retry every self._load_balancing_interval seconds.
                 # Meanwhile this event processor won't lose the partitions it has claimed before.
                 # If it keeps failing, other EventProcessors will start to claim ownership of the partitions
                 # that this EventProcessor is working on. So two or multiple EventProcessors may be working
-                # on the same partition.
+                # on the same partition for a short while.
+                # When owner_levle by default is set, this time duration will be very short.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: typo: owner_level.

and I think the statement could be improved to like:
Setting owner_level would create exclusive connection to the partition which would alleviate duplicate-receiving greatly.

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py [hunk 2]
```diff
@@ -242,24 +242,31 @@ def _create_tasks_for_claimed_ownership(
...
                     '''
                     ownership_manager.get_checkpoints() and ownership_manager.claim_ownership() may raise exceptions
                     when there are load balancing and/or checkpointing (checkpoint_store isn't None).
                     They're swallowed here to retry every self._load_balancing_interval seconds. Meanwhile this event
                     processor won't lose the partitions it has claimed before.
                     If it keeps failing, other EventProcessors will start to claim ownership of the partitions
                     that this EventProcessor is working on. So two or multiple EventProcessors may be working
-                    on the same partition.
+                    on the same partition for a short while.
+                    When owner_levle by default is set, this time duration will be very short.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py]: same to the above

**Pull request merged**